pkg/template/compiled_template_error.go

```diff
@@ -187,9 +187,20 @@ func (CompiledTemplateMultiError) findClosestLine(ct *CompiledTemplate, posLine
 	}
 }
 
-func (CompiledTemplateMultiError) hintMsg(errMsg string) string {
+// hintMsgHelper checks if the operator is present in the code line to set the hint message appropriately
+func (CompiledTemplateMultiError) hintMsgHelper(err CompiledTemplateError, op string) bool {
```

Comment:
👍🏻   The shape of this [extract method](https://refactoring.guru/extract-method) refactor sliced through to capture the lower-level details into here.

💡  for function/method names, we aim to reveal intention. The grammar that does this is almost always a verb or verb phrase.  One approach is to take the Godoc and come up with a succinct phrase that says the same thing (and for unexported functions, it can often mean you don't need to Godoc comment any more!)